src/conv/problem_description.cpp

```diff
@@ -80,6 +108,21 @@ std::string ProblemDescription::GetDirectionStr() const
     return s;
 }
 
+std::string ProblemDescription::GetAlphaBetaCaseStr() const
+{
+    std::string s;
+
+    switch(GetAlphaBetaCase())
+    {
+    case AlphaBetaCase::BILINEAR: s = "Bilinear"; break;
+    case AlphaBetaCase::SCALE: s = "Scale"; break;
+    case AlphaBetaCase::DEFAULT: s = "Default"; break;
+    default: assert(false);
```

Comment:
This should probably return a string like "Invalid" or "Unsupported". It would be much harder to notice an empty string if this happens in release. It is best to check for errors where the check is expected rather than to rely on it happening as a side-effect of string conversion in debug build.

Reply:
when you say check error in ctor, you mean do MIOPEN_THROW inside [ClassifyAlphaBeta](https://github.com/ROCm/MIOpen/blob/d303cb3d70a0f841de91c1f39f9f74c6da6be364/src/include/miopen/conv/problem_description.hpp#L180) instead of returning [ERROR_STATE](https://github.com/ROCm/MIOpen/blob/d303cb3d70a0f841de91c1f39f9f74c6da6be364/src/conv/problem_description.cpp#L93)? That way we don't have to handle default case?

Reply:
I mean in the constructor of problem description, but yes, not having an "error" case if it is not functional is probably better.

Reply:
now doing [MIOPEN_THROW ](https://github.com/ROCm/MIOpen/blob/b444156dc86ba3111ebb9aab0d84a53f5996e2be/src/conv/problem_description.cpp#L118)in default case.